README.md

```diff
@@ -0,0 +1,146 @@
+  10.0.0.1.password: "<ENTER_YOUR_VCENTER_PASSWORD>"
+  1.2.3.4.username: "<ENTER_YOUR_VCENTER_USERNAME>"
+  1.2.3.4.password: "<ENTER_YOUR_VCENTER_PASSWORD>"
+```
+
+Automation tools like a juju charm will need to read these yaml manifest files, manipulate
+its content, and deploy those manifests when any of the **configurable** data is changed.
+
+
+## Supporting Multiple Releases
+Likewise, the projects which release reference manifest files, will also release versions
+of manifests. It's possible for a charm to load all the supported manifest files into a 
+folder structure such the charm supports multiple releases. This library supports this 
+requirements by having the charm store upstream manifest files unchanged in a folder 
+structure like this:
+
+```
+<base_path>
+├── version                  - a file containing the default version
+├── manifests                - a folder containing all the releases
+│   ├── v1.1.10              - a folder matching a configurable version
+│   │   ├── manifest-1.yaml  - any file with a `.yaml` file type
+│   │   └── manifest-2.yaml
+│   ├── v1.1.11
+│   │   ├── manifest-1.yaml
+│   │   └── manifest-2.yaml
+│   │   └── manifest-3.yaml
+```
+
+Key file-heirarchy requirements
+-------------------------------
+|  |  |
+| --- | --- |
+| **$base_path** | A single charm can support multiple manifest releases
+| **version**    | A text file indicating to the library which manifest version is the default when the 'release' config is unspecified |
+| **manifests**  | A folder containing the individual release manifest folders |
+| **$release**   | A folder containing the yaml files of the specific release |
+
+## Sample Usage
+
+Once your charm includes the above manifest file heirachy, your charm will need to define the
+mutations the library should make to the manifests. 
+
+
+```python
+from ops.manifests import Collector, Manifests, ManifestLabel, ConfigRegistry
+
+class ExampleManifests(Manifests):
+    def __init__(self, charm_config):
+        manipulations = [
+            ManifestLabel(self),
+            ConfigRegistry(self),
+            UpdateSecret(self),
+        ]
+        super().__init__("example", "upstream/example", manipulations=manipulations)
+        self.charm_config = charm_config
+
+    @property
+    def config(self) -> Dict:
+        """Returns config mapped from charm config and joined relations."""
+        config = dict(**self.charm_config)
+
+        for key, value in dict(**config).items():
+            if value == "" or value is None:
+                del config[key]  # blank out keys not currently set to something
+
+        config["release"] = config.pop("example-release", None)
+        return config
+
+
+class ExampleCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        # collection of ManifestImpls
+        self.manifests = [ ExampleManifests(self.charm.config) ]
+
+        # create an Actions handler object
```

Comment:
In the collector docstring you refer to these as collector handler/callbacks, but in the readme you refer to them as actions handler/action callbacks.